### PR TITLE
Add InTransfer transaction

### LINKED ISSUE #888: Add InTransferTransaction class
### Description

Extend `BaseTransaction` implement all the specific features of in-transfer transactions.

### PR DESCRIPTION
**Pull request by shuse2**
### Description
Adds InTransfer transaction handling.
Since it's currently disabled, we do not have `create` method for this.

### Review checklist

* The PR resolves #888 
* Merge after #976 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Review by SargeKhan — APPROVED** [review]

File: packages/lisk-transactions/src/transactions/6_in_transfer_transaction.ts [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+			errors.push(
+				new TransactionError(
+					'Recipient id must be empty',
+					this.id,
+					'.recipientId',
+				),
+			);
+		}
+		if (
+			this.recipientPublicKey !== undefined &&
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/src/transactions/6_in_transfer_transaction.ts]: can do `if (this.recipientPublicKey)`. same for `recipientId` check above

File: packages/lisk-transactions/src/transactions/6_in_transfer_transaction.ts [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+					this.id,
+					'.recipientPublicKey',
+				),
+			);
+		}
+
+		if (this.amount.lte(0)) {
+			errors.push(
+				new TransactionError(
+					'Amount must be greather than 0',
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/src/transactions/6_in_transfer_transaction.ts]: typo

File: packages/lisk-transactions/src/transactions/6_in_transfer_transaction.ts [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+		) {
+			throw new Error('Required state does not have valid account type.');
+		}
+
+		const sender = accounts.find(acct => acct.address === this.senderId);
+		if (!sender) {
+			throw new Error('No sender account is found.');
+		}
+
+		// In valid case, transaction should not exist
```
Comment (mitsuaki-u) [inline packages/lisk-transactions/src/transactions/6_in_transfer_transaction.ts]: not sure if this comment is true

**Comment by shuse2** [comment]
@mitsujutsu addressed

**Review by mitsuaki-u — APPROVED** [review]

**Pull request merged**